packages/@lwc/engine-core/src/patches/detect-non-standard-aria.ts

```diff
@@ -0,0 +1,120 @@
+    // Else it might be a light DOM component. Walk up the tree trying to find the owner
+    let parentElement: Element | null = elm;
+    while (!isNull((parentElement = parentElement.parentElement))) {
+        if (isLightningElement(parentElement)) {
+            const vm = getAssociatedVMIfPresent(parentElement);
+            if (!isUndefined(vm)) {
+                return vm;
+            }
+        }
+    }
+    // If we return undefined, it's because the element was rendered wholly outside a LightningElement
+}
+
+function checkAndReportViolation(elm: Element, prop: string) {
+    if (!isLightningElement(elm)) {
+        const vm = findVM(elm);
+
+        if (process.env.NODE_ENV !== 'production') {
+            logWarnOnce(
+                `Element <${elm.tagName.toLowerCase()}> ` +
+                    (isUndefined(vm) ? '' : `owned by <${vm.elm.tagName.toLowerCase()}> `) +
+                    `uses non-standard property "${prop}". This will be removed in a future version of LWC. ` +
+                    `See https://lwc.dev/guide/accessibility#deprecated-aria-reflected-properties`
+            );
+        }
+        report(ReportingEventId.NonStandardAriaReflection, vm);
+    }
+}
+
+function enableDetection() {
+    const { prototype } = Element;
+    for (const prop of NON_STANDARD_ARIA_PROPS) {
+        const descriptor = getOwnPropertyDescriptor(prototype, prop);
+        // The descriptor should exist because the @lwc/aria-reflection polyfill has run by now.
+        // This happens automatically because of the ordering of imports.
+        if (process.env.NODE_ENV !== 'production') {
+            /* istanbul ignore if */
+            if (
+                isUndefined(descriptor) ||
+                isUndefined(descriptor.get) ||
+                isUndefined(descriptor.set)
+            ) {
+                // should never happen
+                throw new Error('detect-non-standard-aria.ts loaded before @lwc/aria-reflection');
+            }
+        }
+        // @ts-ignore
+        const { get, set } = descriptor;
+        defineProperty(prototype, prop, {
+            get() {
+                checkAndReportViolation(this, prop);
+                return get.call(this);
+            },
+            set(val) {
+                checkAndReportViolation(this, prop);
+                return set.call(this, val);
+            },
+            configurable: true,
+            enumerable: true,
+        });
+    }
+}
+
+// No point in running this code if we're not in a browser, or if the global polyfill is not loaded
+if (process.env.IS_BROWSER) {
+    if (!features.DISABLE_ARIA_REFLECTION_POLYFILL) {
+        // Always run detection in dev mode, so we can at least print to the console
+        if (process.env.NODE_ENV !== 'production') {
+            enableDetection();
+        } else {
+            // In prod mode, only enable detection if reporting is enabled
+            onReportingEnabled(enableDetection);
+        }
+    }
+}
```

Comment:
This code potentially adds perf overhead, but I'm pretty sure it should be minimal. Effectively we are just adding getter/setter hooks to the `aria*` props.